bigtable/tableadmin/tableadmin.py

```diff
@@ -0,0 +1,253 @@
+    # [END bigtable_create_family_gc_union]
+
+    # [START bigtable_create_family_gc_intersection]
+    print 'Creating column family cf4 with Intersection GC rule...'
+    # Create a column family with GC policy to drop data that matches
+    # all conditions
+    column_family_id4 = 'cf4'
+    # GC rule: Drop cells older than 5 days AND older than the most
+    # recent 2 versions
+    intersection = [max_age_rule, max_versions_rule]
+    intersection_rule = table.gc_rule_intersection(union)
+    print 'Created Intersection GC Rule.'
+
+    cf4 = table.column_family(column_family_id4, intersection_rule)
+    cf4.create()
+    print 'Created column family cf4 with Intersection GC rule.'
+    # [END bigtable_create_family_gc_intersection]
+
+    # [START bigtable_create_family_gc_nested]
+    print 'Creating column family cf5 with a Nested GC rule...'
+    # Create a column family with nested GC policys.
+    # Create a nested GC rule:
+    # Drop cells that are either older than the 10 recent versions
+    # OR
+    # Drop cells that are older than a month AND older than the
+    # 2 recent versions
+    column_family_id5 = 'cf5'
+    # Drop cells that are either older than the 10 recent versions
+    max_versions_rule1 = table.max_versions_gc_rule(10)
+
+    # Drop cells that are older than a month AND older than
+    # the 2 recent versions
+    max_age = datetime.timedelta(days=30)
+    max_age_rule = table.max_age_gc_rule(max_age)
+    max_versions_rule2 = table.max_versions_gc_rule(2)
+    intersection = [max_age_rule, max_versions_rule2]
+    intersection_rule = table.gc_rule_intersection(intersection)
+
+    # This nesting is done with union rule since it is OR between
+    # the selected rules.
+    nest = [max_versions_rule1, intersection_rule]
+    nested_rule = table.gc_rule_union(nest)
+    print 'Created Nested GC Rule.'
+
+    cf5 = table.column_family(column_family_id5, nested_rule)
+    cf5.create()
+    print 'Created column family cf5 with a Nested GC rule.'
+    # [END bigtable_create_family_gc_nested]
+
+    # [START Printing Column Family and GC Rule for all column families.]
+    print 'Printing Column Family and GC Rule for all column families...'
+    column_families = table.list_column_families()
+    for column_family, gc_rule in sorted(column_families.items()):
+        print "Column Family:", column_family
+        print "GC Rule:"
+        print gc_rule.to_pb()
+    # [END Printing Column Family and GC Rule for all column families.]
+
+
+def exists(instance_obj, table_id):
+    """ Check whether table exists or not.
+
+        Note: This is temporary function as code for this is under development.
+        Once complete, this function would be removed.
+
+    :type instance_obj: Instance Class.
+    :param instance_obj: Instance object.
+
+    :type table_id: str
+    :param table_id: Table id to create table.
+    Returns bool
+    """
+    for table in instance_obj.list_tables():

```

Comment:
I assume list_tables does not return a paginated result, but just want to double check with you.

Reply:
Correct.  Bigtable list RPCs don't support pagination.